### PR TITLE
Add repository IDs and permissions to refreshToken parameters.

### PR DESCRIPTION
**Pull request by emoryruscus**
Add repository IDs and permissions to refreshToken parameters.

### CODE REVIEW

**Comment by bradleyfalzon** [comment]
Are you able to clarify the purpose of these modifications? Just to be clear why it's required. I haven't had a deep look yet, but A) it's unclear B) concerned about potential circular dependencies with `go-github`. Just want to clarify those points before I look further.

File: transport.go [hunk 1]
```diff
@@ -40,6 +45,8 @@ type Transport struct {
 type accessToken struct {
 	Token     string    `json:"token"`
 	ExpiresAt time.Time `json:"expires_at"`
+	Permissions  github.InstallationPermissions `json:"permissions,omitempty"`
+	Repositories []github.Repository            `json:"repositories,emitempty"`
```
Comment (bradleyfalzon) [inline transport.go]: Should this be `omitempty`?

File: transport.go [hunk 2]
```diff
@@ -130,8 +165,37 @@ func (t *Transport) refreshToken(ctx context.Context) error {
...
+}
+
+// GetReadCloser converts a body interface into an io.ReadCloser object.
+func GetReadCloser(i interface{}) (io.ReadCloser, error) {
+	buf, err := GetReadWriter(i)
+	if err != nil {
+		return nil, err
+	}
+
+	all, err := ioutil.ReadAll(buf)
```
Comment (bradleyfalzon) [inline transport.go]: I haven't understood why you're reading `buf` and then passing to `ioutil.NopCloser`. At a (very) cursory glance, couldn't the user just do:

```go
    reader, _ := GetReadWriter(a)
    readCloser, _ := ioutil.NopCloser(reader)
```

And then we could get rid of the entire `GetReadCloser` func?

File: transport_test.go [hunk 3]
```diff
@@ -173,3 +176,72 @@ func TestNew_appendHeader(t *testing.T) {
 		t.Errorf("could not find %v in request's accept headers: %v", myheader, headers["Accept"])
 	}
 }
+
+func TestRefreshTokenWithParameters(t *testing.T) {
+	installationTokenOptions := &github.InstallationTokenOptions{
+		RepositoryIDs: []int64{1234},
+		Permissions: &github.InstallationPermissions{
+			Contents: String("write"),
+			Issues:   String("read"),
```
Comment (bradleyfalzon) [inline transport_test.go]: Have I missed something, or where is the `String` function defined? Are you referring to `github.String()`?

**Comment by emoryruscus** [comment]
You are welcome! Fixed the errors you pointed out. Is there a reason not to import go-github? Seemed like the correct place to update InstallationTokenOptions to me.

File: transport.go [hunk 4]
```diff
@@ -130,8 +165,32 @@ func (t *Transport) refreshToken(ctx context.Context) error {
...
+		err := enc.Encode(i)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return buf, nil
+}
+
+// GetReadCloser converts a body interface into an io.ReadCloser object.
+func GetReadCloser(i interface{}) (io.ReadCloser, error) {
```
[the remark this answers is not in the document]
Comment (emoryruscus) [reply transport.go]: Done! No problem.

**Comment by bradleyfalzon** [comment]
Thanks!

**Comment by emoryruscus** [comment]
My apologies, I think I messed up one thing. Fixed in https://github.com/bradleyfalzon/ghinstallation/pull/25

Fix 'github.com/google/gocmp/cmp' -> 'github.com/google/go-cmp/cmp'

**Pull request merged**